mmpose/datasets/transforms/common_transforms.py

```diff
@@ -873,58 +874,136 @@ class GenerateTarget(BaseTransform):
         - keypoints_visible
         - dataset_keypoint_weights
 
-    Added Keys (depends on the args):
-        - heatmaps
-        - keypoint_labels
-        - keypoint_x_labels
-        - keypoint_y_labels
-        - keypoint_weights
+    Added Keys:
+
+        - The keys of the encoded items from the codec will be updated into
+            the results, e.g. ``'heatmaps'`` or ``'keypoint_weights'``. See
+            the specific codec for more details.
 
     Args:
         encoder (dict | list[dict]): The codec config for keypoint encoding
-        target_type (str): The type of the encoded form of the keypoints.
-            Should be one of the following options:
-
-            - ``'heatmap'``: The encoded should be instance-irrelevant
-                heatmaps and will be stored in ``results['heatmaps']``
-            - ``'multilevel_heatmap'`` The encoded should be a list of
-                heatmaps and will be stored in ``results['heatmaps']``.
-                Note that in this case, ``self.encoder`` should also be
-                a list, and each encoder encodes a single-level heatmaps.
-            - ``'keypoint_label'``: The encoded should be instance-level
-                labels and will be stored in ``results['keypoint_label']``
-            - ``'keypoint_xy_label'``: The encoed should be instance-level
-                labels in x-axis and y-axis respectively. They will be stored
-                in ``results['keypoint_x_label']`` and
-                ``results['keypoint_y_label']``
-            - ``'heatmap+keypoint_label'``: The encoded should be heatmaps and
-                keypoint_labels, will be stored in ``results['heatmaps']``
-                and ``results['keypoint_label']``
+        target_type (str, deprecated): This argument is deprecated and has no
+            effect
         use_dataset_keypoint_weights (bool): Whether use the keypoint weights
             from the dataset meta information. Defaults to ``False``
     """
 
     def __init__(self,
                  encoder: MultiConfig,
-                 target_type: str,
+                 target_type: Optional[str] = None,
+                 multilevel: bool = False,
                  use_dataset_keypoint_weights: bool = False) -> None:
         super().__init__()
+
+        if target_type is not None:
+            warnings.warn(
+                'The argument `target_type` in GenerateTarget is deprecated.'
+                'The target type and encoded keys will be directly')
```

Comment:
Seems like miss some message here: wii be directly updated into the results dict.